olive-editor produces linkage error

## Issue description

Linkage error when doing `nix-env -i olive-editor`

Full log: https://gist.github.com/sshilovsky/980b962d862c636d30056614f410bcd8

Excerpt:
```
g++ -Wl,-O1 -Wl,-rpath,/nix/store/zbxrsrpcac565g0fdc3s1dlz9c0rgcla-qtbase-5.14.2/lib -o olive-editor main.o mainwindow.o project.o effectcontrols.o viewer.o timeline.o sourcetable.o aboutdialog.o timelinewidget.o media.o footage.o sequence.o clip.o config.o newsequencedialog.o viewerwidget.o viewercontainer.o exportdialog.o collapsiblewidget.o panels.o exportthread.o timelineheader.o previewgenerator.o labelslider.o preferencesdialog.o audiomonitor.o undo.o scrollarea.o comboboxex.o colorbutton.o replaceclipmediadialog.o fontcombobox.o keyframeview.o texteditex.o demonotice.o marker.o speeddialog.o mediapropertiesdialog.o projectmodel.o loadthread.o loaddialog.o debug.o path.o linearfadetransition.o transformeffect.o solideffect.o texteffect.o timecodeeffect.o audionoiseeffect.o paneffect.o toneeffect.o volumeeffect.o crossdissolvetransition.o shakeeffect.o exponentialfadetransition.o logarithmicfadetransition.o cornerpineffect.o math.o effect.o effectrow.o effectgizmo.o clipboard.o resizablescrollbar.o sourceiconview.o sourcescommon.o keyframenavigator.o grapheditor.o graphview.o keyframedrawing.o clickablelabel.o keyframe.o rectangleselect.o actionsearch.o embeddedfilechooser.o fillleftrighteffect.o voideffect.o texteditdialog.o debugdialog.o viewerwindow.o projectfilter.o frei0reffect.o effectloaders.o vsthost.o flowlayout.o proxydialog.o proxygenerator.o advancedvideodialog.o cursors.o menuhelper.o global.o focusfilter.o comboaction.o mediaiconservice.o panel.o dropshadoweffect.o renderfunctions.o renderthread.o cacher.o clipqueue.o audio.o clippropertiesdialog.o framebufferobject.o updatenotification.o icons.o doublefield.o fontfield.o effectfield.o colorfield.o stringfield.o boolfield.o combofield.o filefield.o labelfield.o buttonfield.o effectui.o transition.o styling.o undostack.o richtexteffect.o blur.o menu.o mediaimportdata.o autocutsilencedialog.o columnedgridlayout.o qrc_icons.o qrc_internalshaders.o qrc_cursors.o moc_mainwindow.o moc_project.o moc_effectcontrols.o moc_viewer.o moc_timeline.o moc_sourcetable.o moc_aboutdialog.o moc_timelinewidget.o moc_newsequencedialog.o moc_viewerwidget.o moc_viewercontainer.o moc_exportdialog.o moc_collapsiblewidget.o moc_exportthread.o moc_timelineheader.o moc_previewgenerator.o moc_labelslider.o moc_preferencesdialog.o moc_audiomonitor.o moc_comboboxex.o moc_colorbutton.o moc_replaceclipmediadialog.o moc_fontcombobox.o moc_keyframeview.o moc_texteditex.o moc_demonotice.o moc_speeddialog.o moc_mediapropertiesdialog.o moc_projectmodel.o moc_loadthread.o moc_loaddialog.o moc_transformeffect.o moc_solideffect.o moc_texteffect.o moc_timecodeeffect.o moc_audionoiseeffect.o moc_paneffect.o moc_toneeffect.o moc_volumeeffect.o moc_shakeeffect.o moc_cornerpineffect.o moc_effect.o moc_effectrow.o moc_effectgizmo.o moc_resizablescrollbar.o moc_sourceiconview.o moc_sourcescommon.o moc_keyframenavigator.o moc_grapheditor.o moc_graphview.o moc_clickablelabel.o moc_actionsearch.o moc_embeddedfilechooser.o moc_fillleftrighteffect.o moc_voideffect.o moc_texteditdialog.o moc_debugdialog.o moc_viewerwindow.o moc_projectfilter.o moc_frei0reffect.o moc_vsthost.o moc_proxydialog.o moc_proxygenerator.o moc_advancedvideodialog.o moc_menuhelper.o moc_global.o moc_focusfilter.o moc_mediaiconservice.o moc_panel.o moc_renderthread.o moc_cacher.o moc_audio.o moc_clippropertiesdialog.o moc_updatenotification.o moc_doublefield.o moc_fontfield.o moc_effectfield.o moc_stringfield.o moc_filefield.o moc_labelfield.o moc_effectui.o moc_boolfield.o moc_buttonfield.o moc_colorfield.o moc_combofield.o moc_transition.o moc_richtexteffect.o moc_autocutsilencedialog.o moc_columnedgridlayout.o   -L/nix/store/5dpz4gs6qc2ppkan80fi9w461l26fbh1-libGL-1.4.0/lib -L/nix/store/0nmpzffqwzrxwwhwb5r8mbcdvhk1frcp-ffmpeg-full-4.4.1/lib -lavutil -lavformat -lavcodec -lavfilter -lswscale -lswresample /nix/store/chqs8c5r6h9bl9a39jdb8bmhbcvvl9q3-qtmultimedia-5.14.2/lib/libQt5Multimedia.so /nix/store/zbxrsrpcac565g0fdc3s1dlz9c0rgcla-qtbase-5.14.2/lib/libQt5OpenGL.so /nix/store/wkdaf51wi98f60k5r4wqhw54irmgzmn3-qtsvg-5.14.2/lib/libQt5Svg.so /nix/store/zbxrsrpcac565g0fdc3s1dlz9c0rgcla-qtbase-5.14.2/lib/libQt5Widgets.so /nix/store/zbxrsrpcac565g0fdc3s1dlz9c0rgcla-qtbase-5.14.2/lib/libQt5Gui.so /nix/store/zbxrsrpcac565g0fdc3s1dlz9c0rgcla-qtbase-5.14.2/lib/libQt5Network.so /nix/store/zbxrsrpcac565g0fdc3s1dlz9c0rgcla-qtbase-5.14.2/lib/libQt5Core.so -L/nix/store/f64g88aqwq3y7zjxrish51i6cb2a5fpa-libglvnd-1.4.0/lib -lGL -lpthread   
/nix/store/bffjk93gw5y9z3l90j56zw87g0yjgwl4-binutils-2.38/bin/ld: /nix/store/xjbz96yj0wqjj3kff2c4yv765lwa6bi2-zimg-3.0.4/lib/libzimg.so.2: undefined reference to `std::__exception_ptr::exception_ptr::_M_release()@CXXABI_1.3.13'
/nix/store/bffjk93gw5y9z3l90j56zw87g0yjgwl4-binutils-2.38/bin/ld: /nix/store/xjbz96yj0wqjj3kff2c4yv765lwa6bi2-zimg-3.0.4/lib/libzimg.so.2: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::reserve()@GLIBCXX_3.4.29'
/nix/store/bffjk93gw5y9z3l90j56zw87g0yjgwl4-binutils-2.38/bin/ld: /nix/store/0nmpzffqwzrxwwhwb5r8mbcdvhk1frcp-ffmpeg-full-4.4.1/lib/libavfilter.so: undefined reference to `std::__throw_bad_array_new_length()@GLIBCXX_3.4.29'
/nix/store/bffjk93gw5y9z3l90j56zw87g0yjgwl4-binutils-2.38/bin/ld: /nix/store/xjbz96yj0wqjj3kff2c4yv765lwa6bi2-zimg-3.0.4/lib/libzimg.so.2: undefined reference to `std::__exception_ptr::exception_ptr::_M_addref()@CXXABI_1.3.13'
collect2: error: ld returned 1 exit status
make: *** [Makefile:851: olive-editor] Error 1
```


### Steps to reproduce

`nix-env -i olive-editor`

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.46, NixOS, 22.05 (Quokka), 22.05.915.9ff91ce2e4c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"nixos-22.05"`
 - channels(sh): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

@balsoft

--- a/pkgs/applications/video/olive-editor/default.nix
+++ b/pkgs/applications/video/olive-editor/default.nix
@@ -1,8 +1,8 @@
-{ lib, stdenv, fetchFromGitHub, pkg-config, which, qmake, mkDerivation,
-  qtmultimedia, wrapQtAppsHook, frei0r, opencolorio_1, ffmpeg-full,
-  CoreFoundation }:
+{ lib, stdenv, fetchFromGitHub
+, pkg-config, which, qmake, wrapQtAppsHook
+, qtmultimedia, frei0r, opencolorio_1, ffmpeg-full, CoreFoundation }:
 
-mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "olive-editor";
   version = "0.1.2";
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -9167,8 +9167,9 @@ with pkgs;
 
   ola = callPackage ../applications/misc/ola { };
 
-  olive-editor = libsForQt514.callPackage ../applications/video/olive-editor
-    { stdenv = gcc10StdenvCompat;     inherit (darwin.apple_sdk.frameworks) CoreFoundation; };
+  olive-editor = libsForQt514.callPackage ../applications/video/olive-editor {
+    inherit (darwin.apple_sdk.frameworks) CoreFoundation;
+  };
 
   ombi = callPackage ../servers/ombi { };
 

